### PR TITLE
Add migration adding contact_type to Contact

### LINKED ISSUE #300: Feature Request: Contact Identification
Stacey J. wants to create three categories for contacts listed.
- Emergency Contact
- Reference
- General Contact

This should be required and be multi-select or checkboxes (as your programming brain desires). Someone entered must be at least one of these, and could be more than that. Not a priority.

### PR DESCRIPTION
**Pull request by adelevie**
fixes #300

### CODE REVIEW

File: app/models/contact.rb [hunk 1]
```diff
@@ -11,6 +11,24 @@ class Contact < ActiveRecord::Base
...
+    {
+      name: "Reference",
+      value: "reference"
+    },
+    {
+      name: "General Contact",
+      value: "general_contact"
+    }
+  ]
+  #enum contact_type: CONTACT_TYPES.map {|ct| ct[:value]}
```
Comment (jrunningen) [inline app/models/contact.rb]: Looks like you left some commented-out code in this pull request.  Please take that out.

Comment (adelevie) [inline app/models/contact.rb]: removed

**Comment by adelevie** [comment]
Not sure why I'm getting 18 test failures:

```
➜  districthousing git:(add_contact_type_to_contact) ✗ bundle exec rake test
Run options: --seed 2898

# Running:

..F..........F..FFFFFFFFFFFFF.......................FF.......................................F.

Finished in 6.318880s, 15.0343 runs/s, 49.0593 assertions/s.

  1) Failure:
ApplicantFactoryTest#test_create_sample_applicant [/Users/alandelevie/districthousing/test/unit/applicant_factory_test.rb:5]:
{:"contacts.contact_type"=>["is not included in the list"], :contacts=>["is invalid"]}


  2) Failure:
ContactTest#test_valid [/Users/alandelevie/districthousing/test/models/contact_test.rb:11]:
{:contact_type=>["is not included in the list"]}


  3) Failure:
FieldFillingTest#test_fills_addresses [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


  4) Failure:
FieldFillingTest#test_fills_basic_personal_information [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


  5) Failure:
FieldFillingTest#test_fills_boolean_fields_in_a_way_that_makes_sense [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


  6) Failure:
FieldFillingTest#test_fills_contact_information [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


  7) Failure:
FieldFillingTest#test_fills_criminal_history_information [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


  8) Failure:
FieldFillingTest#test_fills_ethnicities [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


  9) Failure:
FieldFillingTest#test_fills_household_member_information [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


 10) Failure:
FieldFillingTest#test_fills_income_information [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


 11) Failure:
FieldFillingTest#test_fills_landlord_information [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


 12) Failure:
FieldFillingTest#test_fills_races [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


 13) Failure:
FieldFillingTest#test_fills_residences [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


 14) Failure:
FieldFillingTest#test_nationality_and_citizenship_are_the_same [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


 15) Failure:
FieldFillingTest#test_other_fields [/Users/alandelevie/districthousing/test/unit/field_filling_test.rb:8]:
{:contacts=>["is invalid"]}


 16) Failure:
SalesforceApplicantsControllerTest#test_sync [/Users/alandelevie/districthousing/test/controllers/salesforce_applicants_controller_test.rb:87]:
Expected: "New first name"
  Actual: "One"


 17) Failure:
SalesforceApplicantsControllerTest#test_sync_all [/Users/alandelevie/districthousing/test/controllers/salesforce_applicants_controller_test.rb:75]:
Expected: "Existing applicant"
  Actual: "One"


 18) Failure:
ContactsControllerTest#test_new [/Users/alandelevie/districthousing/test/controllers/contacts_controller_test.rb:10]:
"Contact.count" didn't change by 1.
Expected: 3
  Actual: 2

95 runs, 310 assertions, 18 failures, 0 errors, 0 skips
```

**Comment by adelevie** [comment]
any ideas, @jrunningen?

File: app/models/contact.rb [hunk 2]
```diff
@@ -11,6 +11,23 @@ class Contact < ActiveRecord::Base
...
+      name: "Reference",
+      value: "reference"
+    },
+    {
+      name: "General Contact",
+      value: "general_contact"
+    }
+  ]
+  
+  validates :contact_type, inclusion: {in: CONTACT_TYPES.map {|ct| ct[:value]}}
```
Comment (jrunningen) [inline app/models/contact.rb]: It looks like this enum validation won't permit a blank contact type.  I'm guessing that's the reason for the first test failure.

Comment (adelevie) [inline app/models/contact.rb]: Added an `allow_nil` option, tests pass locally. just let travis do its thing ...

**Pull request merged**